### ISSUE TITLE
DatePicker Accessibility - sticky keys can break out of popup

### ISSUE THREAD

**Issue by petematt** [angular/components#8484]
#### Bug, feature request, or proposal:
The datepicker popup does not trap focus inside it when chromevox sticky keys mode is enabled.

#### What is the expected behavior?
Arrow keys in sticky mode should behave like tabs and keep focus inside the popup.

#### What is the current behavior?
When selecting a date in the datepicker popup with sticky keys enabled, you can hit the arrow keys and escape the dialog box entirely.

#### What are the steps to reproduce?
1) Navigate to http://plnkr.co/edit/LzF48jm7ipwKdZWWxTwR
2) Enable chromevox and enable sticky keys.
3) Open the date picker.
4) Use the arrows to navigate outside the popup.

#### What is the use-case or motivation for changing an existing behavior?
Accessibility

#### Which versions of Angular, Material, OS, TypeScript, browsers are affected?
Version: 5.0.0-rc0-541a95e

**Comment by jelbourn** [comment]
Is this ChromeVox on a Chomebook for ChromeVox the browser extension? If it's the latter, we specifically don't support it because it is not a useful end-user screen reader.

**Comment by jelbourn** [comment]
Also are you able to reproduce this in any other screen-reader? (NVDA, JAWS, VoiceOver)

**Comment by petematt** [comment]
I've verified this on a chromebook.  I will check with other screen readers and see if the results are consistent.

**Comment by zarend** [comment]
We've made some fixes to the focus trapping in the datepicker dialog lately, so this might have been fixed by #24300. I can verify this next time I have my chromebook with me.

**Comment by zarend** [comment]
I cannot reproduce this issue. Perhaps it was fixed by the recent work we do on the focus trapping.

If there are still problems, please reopen this issue or file a new bug report.
